graphql_compiler/macros/macro_edge/validation.py

```diff
@@ -61,6 +65,45 @@ def _validate_macro_ast_with_macro_directives(schema, ast, macro_directives):
                 .format(directive_definition.name, len(macro_data)))
 
 
+def _validate_macro_ast_directives(ast, inside_fold_scope=False):
+    """Check that the macro is using non-macro direcives properly."""
+    subselection_inside_fold_scope = inside_fold_scope
+    directives_with_no_restrictions = {
+        FilterDirective.name,
+        TagDirective.name,
+        OptionalDirective.name,
+        RecurseDirective.name,
+        MacroEdgeDefinitionDirective.name,
+    }
+    for directive in ast.directives:
+        name = directive.name.value
+        if name in directives_with_no_restrictions:
+            pass
+        elif name == OutputDirective.name:
+            raise GraphQLInvalidMacroError(u'Macros must not have outputs. '
```

Comment:
These cases are very similar, what about generalizing and using the name of the directive in the error message? 
ie: `raise GraphQLInvalidMacroError(u'Macros are not allowed to use the "{}" directive. Found usage {}.'.format(name, directive)`